domain-models-runtime/src/main/java/org/folio/rest/tools/utils/TenantLoading.java

```diff
@@ -392,9 +392,8 @@ private static void loadURL(Map<String, String> headers, URL url,
    * @param vertx Vertx handle to be used (for spawning HTTP clients)
    * @return Success or failure
    */
-  public Future<Void> perform(TenantAttributes ta, Map<String, String> headers,
-      Context context) {
-    return perform0(ta, headers, context.owner()).mapEmpty();
+  public Future<Void> perform(TenantAttributes ta, Map<String, String> headers, Vertx vertx) {
+    return perform0(ta, headers, vertx).mapEmpty();
```

Comment:
The `Future<Void> perform` method should be designed to match the types used in TenantRef loadData method:
```
Future<Void> loadData(TenantAttributes attributes, String tenantId,
                      Map<String, String> headers, Context vertxContext)
```
The Future type should be Void and not Integer.
It should take a Context and not a Vertx. (The javadoc in line 392 should be changed.)
This will result in the most straightforward usage as explained in the documentation:
https://github.com/folio-org/raml-module-builder/pull/807/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R1377
https://github.com/folio-org/raml-module-builder/pull/807/files#diff-95cca51db9a9a6cf4c286410f08fe5eec3c68456869faf08739d8b0235fdedd9R71
We don't want all the modules using the TenantLoading loadData interface to duplicate the code that extracts the Vertx from vertxContext. This task should be handled by the perform method. The other perform method that has been used for postTenant is legacy and is not changed to avoid a breaking change: it takes a `Promise<Integer>` (not `Promise<Void>`) and takes a Vertx (not a vertx Context).

Reply:
OK. Fine as far as Context goes.. I'll change it to take that.. But isn't it bad if we throw away the Integer in the result. At least it allows a user of the API to determine the number of items loaded.

Reply:
I've made the requested change: Not throwing away the number of records loaded but returning them as `Future<Integer>`. There isn't any module that uses this number yet, though.